source/internal.d.ts

```diff
@@ -328,3 +408,283 @@ IsPrimitive<Object>
+```
+Not<true>
+//=> false
+
+Not<false>
+//=> true
+```
+*/
+export type Not<A extends boolean> = A extends true
+	? false
+	: A extends false
+		? true
+		: boolean;
+
+/**
+Returns the maximum value from a tuple of integers.
+
+Note:
+- float number is not supported.
+
+@example
+```
+ArrayMax<[1, 2, 5, 3]>;
+//=> 5
+
+ArrayMax<[1, 2, 5, 3, 99, -1]>
+//=> 99
+```
+*/
+export type ArrayMax<A extends number[], Result extends number = NegativeInfinity> = A extends [infer F extends number, ...infer R extends number[]]
+	? Gt<F, Result> extends true
+		? ArrayMax<R, F>
+		: ArrayMax<R, Result>
+	: Result;
+
+/**
+Returns the minimal value from a tuple of integers.
+
+@example
+```
+ArrayMin<[1, 2, 5, 3]>
+//=> 1
+
+ArrayMin<[1, 2, 5, 3, -5]>
+//=> -5
+```
+*/
+export type ArrayMin<A extends number[], Result extends number = PositiveInfinity> = A extends [infer F extends number, ...infer R extends number[]]
+	? Lt<F, Result> extends true
+		? ArrayMin<R, F>
+		: ArrayMin<R, Result>
+	: Result;
+
+/**
+Returns the absolute value of a given value.
+
+@example
+```
+NumberAbsolute<-1>
+//=> 1
+
+NumberAbsolute<1>
+//=> 1
+
+NumberAbsolute<NegativeInfinity>
+//=> PositiveInfinity
+```
+*/
+export type NumberAbsolute<N extends number> = `${N}` extends `-${infer StringPositiveN}` ? StringToNumber<StringPositiveN> : N;
+
+/**
+Returns a boolean for whether A > B.
+
+@example
+```
+Gt<1, -5>
+//=> true
+
+Gt<1, 1>
+//=> false
+
+Gt<1, 5>
+//=> false
+```
+*/
+// TODO: Support large integer.
+export type Gt<A extends number, B extends number> = number extends A | B
+	? boolean
+	: [
+		IsEqual<A, PositiveInfinity>, IsEqual<A, NegativeInfinity>,
+		IsEqual<B, PositiveInfinity>, IsEqual<B, NegativeInfinity>,
+	] extends infer R extends [boolean, boolean, boolean, boolean]
+		? Or<
+		And<IsEqual<R[0], true>, IsEqual<R[2], false>>,
+		And<IsEqual<R[3], true>, IsEqual<R[1], false>>
+		> extends true
+			? true
+			: Or<
+			And<IsEqual<R[1], true>, IsEqual<R[3], false>>,
+			And<IsEqual<R[2], true>, IsEqual<R[0], false>>
+			> extends true
+				? false
+				: true extends R[number]
+					? false
+					: [IsNegative<A>, IsNegative<B>] extends infer R extends [boolean, boolean]
+						? [true, false] extends R
+							? false
+							: [false, true] extends R
+								? true
+								: [false, false] extends R
+									? PositiveNumericStringGt<`${A}`, `${B}`>
+									: PositiveNumericStringGt<`${NumberAbsolute<B>}`, `${NumberAbsolute<A>}`>
+						: never
+		: never;
+
+/**
+Returns a boolean for whether A >= B.
+
+@example
+```
+Gte<1, -5>
+//=> true
+
+Gte<1, 1>
+//=> true
+
+Gte<1, 5>
+//=> false
+```
+*/
+export type Gte<A extends number, B extends number> = number extends A | B
+	? boolean
+	: A extends B ? true : Gt<A, B>;
+
+/**
+Returns a boolean for whether A < B.
+
+@example
+```
+Lt<1, -5>
+//=> false
+
+Lt<1, 1>
+//=> false
+
+Lt<1, 5>
+//=> true
+```
+*/
+
+export type Lt<A extends number, B extends number> = number extends A | B
+	? boolean
```

Comment:
let's talk about the returns should be `never` or `boolean`

Reply:
see `StartsWith` comment